packages/amplify-provider-awscloudformation/lib/push-resources.js

```diff
@@ -199,9 +200,9 @@ function validateCfnTemplates(context, resourcesToBeUpdated) {
     const { category, resourceName } = resourcesToBeUpdated[i];
     const backEndDir = context.amplify.pathManager.getBackendDirPath();
     const resourceDir = path.normalize(path.join(backEndDir, category, resourceName));
-    const files = fs.readdirSync(resourceDir);
-    // Fetch all the Cloudformation templates for the resource (can be json or yml)
-    const cfnFiles = files.filter(file => file.indexOf('template') !== -1 && file.indexOf('.') !== 0);
+    const cfnFiles = glob.sync('*template*.+(yaml|yml|json)', {
```

Comment:
Do we want to support files which have something else after template like `myApi.template.first.json`
Can this be pulled to a const.
```suggestion
   const CFN_GLOB_PATTERN = '*template*.+(yaml|yml|json)';
    const cfnFiles = glob.sync(CFN_GLOB_PATTERN , {
```

Reply:
Currently we don't have any requirement on this, this changed followed the same pattern what we had before, which allows this, later we can revisit it and even just enable anything with those extensions except parameters.json (already included in the update) if we want to remote the 'template' restriction later.
Good catch, moved to constant for all codepaths